stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/ChatNotifications.kt

```diff
@@ -38,7 +39,7 @@ internal interface ChatNotifications {
     fun onPushMessage(message: PushMessage, pushNotificationReceivedListener: PushNotificationReceivedListener)
     fun onNewMessageEvent(newMessageEvent: NewMessageEvent)
     suspend fun onLogout(flushPersistence: Boolean)
-    fun displayNotification(channel: Channel, message: Message)
+    fun displayNotification(type: String, channel: Channel, message: Message)
```

Comment:
Instead of an string defining the type of PN, Why don't we use a sealed class??
It will allow us in the future to provide different data for every PN type without a breaking change

Reply:
The problem by keeping a "single data class" is that it force us to have always the same properties.
Just think on a PN type "Reaction", would be good to have the message where the reaction was added, the channel where the message is present, but we should know Who and Which reaction was added... The information could be inside of the Message itself, but we need to do extra logic to try to identify which reaction was the one we are receiving this PN from.
Instead of that, if this info is received from the PN Payload and mapped to a new `Notification.Reaction` class it would be easier to handle.
```
data class Reaction(val reaction: Reaction, val from: User, val message: Message, val channel: Channel) : Notification
```